arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -600,6 +600,11 @@ void HttpCommTask<T>::sendResponse(std::unique_ptr<GeneralResponse> baseRes,
   // will add CORS headers if necessary
   this->finishExecution(*baseRes, _origin);
 
+  // handle response code 204 No Content
+  if (response.responseCode() == rest::ResponseCode::NO_CONTENT) {
+    response.clearBody();  // in non maintainer build clear the body
```

Comment:
I guess we are clearing the body in both maintainer and non-maintainer mode. Or am I missing something?
```suggestion
    response.clearBody();
```